plugin/lib/class-php-compatibility-checker.php

```diff
@@ -109,66 +108,105 @@ public function init() {
 	}
 
 	/**
-	 * Generates a list of directories to scan and populate the queue.
+	 * Check if a plugin should be excluded from scans.
 	 *
-	 * @since  1.0.0
+	 * @param array $plugin_data Plugin data.
+	 * @see https://developer.wordpress.org/reference/functions/get_plugin_data/
+	 * @return boolean
 	 */
-	public function generate_directory_list() {
-		if ( ! function_exists( 'get_plugins' ) ) {
+	public function exclude_plugin( $plugin_data ) {
+		$excluded_plugins = array( 'PHP Compatibility Checker', 'Hello Dolly' );
+		$excluded_plugins = apply_filters( 'phpcompat_excluded_plugins', $excluded_plugins );
+
+		return in_array( $plugin_data['Name'], $excluded_plugins, true );
+	}
 
+	/**
+	 * Get all plugins that should be scanned.
+	 *
+	 * @return array
+	 */
+	public function get_plugins_to_scan() {
+		if ( ! function_exists( 'get_plugins' ) ) {
+			// phpcs:ignore PEAR.Files.IncludingFile.UseIncludeOnce -- strict requirement
 			require_once ABSPATH . 'wp-admin/includes/plugin.php';
 		}
 
-		$plugin_base = dirname( $this->base ) . DIRECTORY_SEPARATOR;
-		$all_plugins = get_plugins();
+		$plugins = get_plugins();
 
-		$update_plugins = get_site_transient( 'update_plugins' );
-		//$slug_name[] = plugin_basename();
-
-		foreach ( $all_plugins as $k => $v ) {
-			// Exclude some plugins.
-			if ( 'PHP Compatibility Checker' === $v['Name'] ) {
-				continue;
-			}
-			if ( 'Hello Dolly' === $v['Name'] ) {
-				continue;
+		// Exclude some plugins.
+		$plugins = array_filter(
```

Comment:
Wondering if we should check if we actually have some plugins after this before proceeding?  The `get_plugins` call above should always return at least this plugin but someone could filter out all other plugins after this step. Not sure it would throw any errors but there's no need to run any additional functionality if we don't have any plugins to scan. Probably an edge case (since why would you have this plugin if you don't have other plugins installed) but might be worth accounting for

Reply:
Thank you for noticing that, updated in https://github.com/10up/phpcompat/pull/2/commits/60037a1be9bf08ba3b5ba426ddef259368d72586